src/System.Management.Automation/engine/Modules/SwitchProcessCommand.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Runtime.InteropServices;
+
+using Dbg = System.Management.Automation.Diagnostics;
+
+#if UNIX
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Implements a cmdlet that allows use of execv API.
+    /// </summary>
+    [Experimental(ExperimentalFeature.PSExecFeatureName, ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Switch, "Process", HelpUri = "https://go.microsoft.com/fwlink/?linkid=2181448")]
+    public sealed class SwitchProcessCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Get or set the command and arguments to replace the current pwsh process.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = false, ValueFromPipeline = true, ValueFromRemainingArguments = true)]
+        public string[] WithCommand { get; set; } = Array.Empty<string>();
+
+        /// <summary>
+        /// For each record, execute it, and push the results into the success stream.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            if (WithCommand.Length == 0)
+            {
+                return;
+            }
+
+            // execv requires command to be full path so resolve command to first match
+            var command = this.SessionState.InvokeCommand.GetCommand(WithCommand[0], CommandTypes.Application);
+            if (command is null)
+            {
+                ThrowTerminatingError(
+                    new ErrorRecord(
+                        new CommandNotFoundException(
+                            string.Format(
+                                System.Globalization.CultureInfo.InvariantCulture,
+                                CommandBaseStrings.NativeCommandNotFound,
+                                command
+                            )
+                        ),
+                        "CommandNotFound",
+                        ErrorCategory.InvalidArgument,
+                        WithCommand
+                    )
+                );
+            }
+
+            var execArgs = new string?[WithCommand.Length + 1];
+
+            // execv convention is the first arg is the program name
+            execArgs[0] = command.Name;
+
+            for (int i = 1; i < WithCommand.Length; i++)
+            {
+                execArgs[i] = WithCommand[i].ToString();
```

Comment:
`WithCommand` is already a `string[]`, so no need to call `ToString` on its elements.
```suggestion
                execArgs[i] = WithCommand[i];
```